test/e2e/pyrolisis_test.go

```diff
@@ -164,6 +172,49 @@ func (p *PyrolisisTests) CleanupBackupStorageContainers(t *testing.T) {
 	}
 }
 
+func (p *PyrolisisTests) CleanupSnapshots(t *testing.T) {
+	storageClient, err := openstack.NewBlockStorageV3(p.OpenStack.Provider, gophercloud.EndpointOpts{})
+	require.NoError(t, err, "There should be no error while creating storage client")
+
+	project, err := tokens.Get(p.OpenStack.Identity, p.OpenStack.Provider.Token()).ExtractProject()
+	require.NoError(t, err, "There should be no error while extracting the project")
+
+	listOpts := snapshots.ListOpts{
+		TenantID: project.ID,
+	}
+
+	allPages, err := snapshots.List(storageClient, listOpts).AllPages()
+	require.NoError(t, err, "There should be no error retieving all snapshot pages")
+
+	allSnapshots, err := snapshots.ExtractSnapshots(allPages)
+	require.NoError(t, err, "There should be no error extracting all snapshots")
+
+	for _, snapshot := range allSnapshots {
+
+		volume, volumeErr := volumes.Get(storageClient, snapshot.VolumeID).Extract()
```

Comment:
instead of looking up metadata on the volume, use the metadata on the snapshot: e.g. `cinder.csi.openstack.org/cluster: e2e-gzd78-d93ad03198be4783befe14afad24c878`

The check should be something like `strings.HasPrefix("e2e-")`

Reply:
Changed. Ready for merge?